### PR TITLE
New unified methods: fetchLiquidations and fetchMyLiquidations

### PR DESCRIPTION
**Pull request by Dan-krm**
Added new unified methods `fetchLiquidations` and `fetchMyLiquidations` to multiple exchanges:

- [x] Binance
- [x] Bitmart
- [x] Bitmex
- [x] Deribit
- [x] Gate
- [x] Huobi

### huobi fetchLiquidations:
```
huobi.fetchLiquidations (BTC/USDT:USDT)
2023-10-11T05:15:41.625Z iteration 0 passed in 312 ms

       symbol | contracts | contractSize |   price | baseValue |  quoteValue |     timestamp |                 datetime
-----------------------------------------------------------------------------------------------------------------------
BTC/USDT:USDT |       985 |        0.001 | 26979.9 |     0.985 |  26575.2015 | 1696994623858 | 2023-10-11T03:23:43.858Z
BTC/USDT:USDT |         4 |        0.001 | 26978.5 |     0.004 |     107.914 | 1696994643655 | 2023-10-11T03:24:03.655Z
BTC/USDT:USDT |         2 |        0.001 | 26978.9 |     0.002 |     53.9578 | 1696994645661 | 2023-10-11T03:24:05.661Z
...
50 objects
```

### CODE REVIEW

File: ts/src/binance.ts [hunk 1]
```diff
@@ -9091,4 +9093,215 @@ export default class binance extends Exchange {
...
+            request['startTime'] = since;
+        }
+        if (limit !== undefined) {
+            if (market['spot']) {
+                request['size'] = limit;
+            } else {
+                request['limit'] = limit;
+            }
+        }
+        const until = this.safeIntegerN (params, [ 'until', 'till', 'endTime' ]);
```
Comment (carlosmiei) [inline ts/src/binance.ts:9128]: we can use the `this.handleUntilOption` to simplify this logic

File: ts/src/binance.ts [hunk 2]
```diff
@@ -9091,4 +9093,215 @@ export default class binance extends Exchange {
...
+         * @see https://binance-docs.github.io/apidocs/futures/en/#user-39-s-force-orders-user_data
+         * @see https://binance-docs.github.io/apidocs/delivery/en/#user-39-s-force-orders-user_data
+         * @param {string} symbol unified CCXT market symbol
+         * @param {int} [since] the earliest time in ms to fetch liquidations for
+         * @param {int} [limit] the maximum number of liquidation structures to retrieve
+         * @param {object} [params] exchange specific parameters for the binance api endpoint
+         * @param {int} [params.until] timestamp in ms of the latest liquidation
+         * @returns {object} an array of [liquidation structures]{@link https://github.com/ccxt/ccxt/wiki/Manual#liquidation-structure}
+         */
+        await this.loadMarkets ();
```
Comment (carlosmiei) [inline ts/src/binance.ts:9112]: also, we can add pagination here, probably the incremental works better in this case

File: ts/src/binance.ts [hunk 3]
```diff
@@ -9091,4 +9093,211 @@ export default class binance extends Exchange {
             'info': interest,
         };
     }
+
+    async fetchMyLiquidations (symbol: string, since: Int = undefined, limit: Int = undefined, params = {}) {
```
Comment (carlosmiei) [inline ts/src/binance.ts:9097]: also, symbol is not mandatory here, so we should update the method's signature 
image

File: ts/src/binance.ts [hunk 4]
```diff
@@ -9091,4 +9093,211 @@ export default class binance extends Exchange {
...
+            if (market['spot']) {
+                request['size'] = limit;
+            } else {
+                request['limit'] = limit;
+            }
+        }
+        [ request, params ] = this.handleUntilOption ('endTime', request, params);
+        let response = undefined;
+        if (market['spot']) {
+            response = await this.sapiGetMarginForceLiquidationRec (this.extend (request, params));
```
Comment (carlosmiei) [inline ts/src/binance.ts:9131]: spot seems to accept a different symbol key 
image

File: ts/src/binance.ts (hunk 2 above)
Comment (Dan-krm) [reply ts/src/binance.ts:9113]: Sounds good, I added it for spot only on binance because it's the only endpoint that has a pagination related param

Comment (Dan-krm) [reply ts/src/binance.ts:9113]: @carlosmiei Which pagination method should we use with deribit and its `continuation` param?
https://docs.deribit.com/#public-get_last_settlements_by_currency
https://docs.deribit.com/#private-get_settlement_history_by_instrument

Comment (carlosmiei) [reply ts/src/binance.ts:9124]: @Dan-krm cursor like we do at bybit, but I will add it nw

**Comment by carlosmiei** [comment]
LGTM

**Pull request merged**